### ISSUE TITLE
Bug: Column Headers not Horizontally Aligned

### ISSUE THREAD

**Issue by jim256** [hankinsoft/SQLPro#292]
Sometimes this will happen:
<img width="248" alt="image" src="https://user-images.githubusercontent.com/15026225/64985320-d7896f80-d881-11e9-81a1-767344c5b10d.png">

I haven't noticed a common denominator. In this particular case, I had the app in full screen. When I switched back from this desktop where I'm typing this to the desktop with SQLPro Studio on it, it was fixed.

**Comment by hankinsoft** [comment]
Hi,
Just to confirm this is a graphical drawing glitch with the column title? Not a horizontal alignment issue (just making sure I understand). Also, do you know if the columns were displayed before you maximized the window, or after? Thanks!

**Comment by jim256** [comment]
I think so.

As for whether it was maximized beforehand, I believe so, but I'm not 100% sure right now. I'll keep an eye out for when it happens again.

**Comment by jim256** [comment]
Just happened again. I was already in full-screen mode.

**Comment by hankinsoft** [comment]
Thanks! I’ll see if I can reproduce it.

**Comment by hankinsoft** [comment]
This might be fixed in https://d3fwkemdw8spx3.cloudfront.net/studio/SQLProStudio.2019.09.20.app.zip. I wasn't able to reproduce, so I'm not 100% sure, but I did make a few improvements around table header drawing. Please let me know if you continue to see it.

**Comment by jim256** [comment]
Just saw it again in the new build (110546)

<img width="255" alt="image" src="https://user-images.githubusercontent.com/15026225/65353564-b0e17680-dbaa-11e9-8124-355ada1212c6.png">

I was in full-screen mode again. I'll try staying out of full-screen mode and see if it continues to happen

**Comment by jim256** [comment]
@hankinsoft 
Darn, looks like that last build didn't quite get it. Just saw it when the app was not in full screen (and hadn't been since launch):
<img width="444" alt="image" src="https://user-images.githubusercontent.com/15026225/65364089-ca93b580-dbcc-11e9-8438-3793a9202cfc.png">

**Comment by hankinsoft** [comment]
Thanks for letting me know. I’ll investigate this further.

**Comment by hankinsoft** [comment]
Can you give https://d3fwkemdw8spx3.cloudfront.net/studio/SQLProStudio.2019.09.24.app.zip? It MIGHT fix it (still haven't been able to reproduce - let me know if that fixes it).

**Comment by jim256** [comment]
Hey @hankinsoft , I've got bad news :(

<img width="558" alt="image" src="https://user-images.githubusercontent.com/15026225/65557915-2bcfc780-def2-11e9-91c5-8b5abdf3a49b.png">

I think this is the worst instance I've personally seen

**Comment by jim256** [comment]
FWIW, looks like this is present in build 110553

**Comment by hankinsoft** [comment]
Please let me know if you still see this in https://d3fwkemdw8spx3.cloudfront.net/studio/SQLProStudio.2019.09.27.app.zip.

**Comment by jim256** [comment]
Darn. Still seeing it.

**Comment by hankinsoft** [comment]
Hi,
Still trying to figure this out.

Two questions if you get the chance:
1. If you resize an individual column, does it fix this?
2. If you resize the window, does it fix this?

Thanks!

**Comment by jim256** [comment]
I'm sorry to have brought up such a challenging issue!

I'll definitely test these out the next time I see this.

**Comment by jim256** [comment]
@hankinsoft:

1. Yes
2. Yes

Also:
- Toggling full-screen mode still fixes it
- If I change desktops (switch to a full-screen app) and come back, it'll be fixed. This works whether SQLPro Studio is full screen or not.
- I just tried scrolling to the right (the columns with the issue in my current results are off the screen to the right) very rapidly as soon as the query finished, one of the columns wasn't having the issue. I was able to repeat this 4x in a row. Then I came here, typed a bit, and now I can't reproduce.
- I just noticed that if I re-query after "fixing" a column via manual resize, that same column will have the issue in the same spot again and again.

For example, the column from this comment is still doing it:
https://github.com/hankinsoft/SQLPro/issues/292#issuecomment-533681679

**Comment by jim256** [comment]
A few more notes:
- I'm starting to think it may only happen if there are more columns than there is space to display them without needing horizontal scrolling.
- You know how sometimes it will display results before they're totally correctly drawn (about 50% of the left-most part of the first column is cut off), and then scoot all of the columns to the right a little bit when it finishes fully? I just noticed that one column had the issue, and then after the usual ~1 second delay when the columns all got scooted, it got fixed.

**Comment by hankinsoft** [comment]
Hi! Thanks for that. I was able to reproduce and am pretty sure its all fixed up now based on this build: https://d3fwkemdw8spx3.cloudfront.net/studio/SQLProStudio.2019.40.app.zip.

**Comment by jim256** [comment]
Glad you were able to reproduce! Good news. Haven't seen it lately. I'll keep this open for a few more days while I keep an eye out for it's happening again, then close it if I don't see it by 10/17.

**Comment by jim256** [comment]
Sorry for the delay here. For some reason, Snowflake has been super slow with any desktop client for me lately, so I haven't been using SQLPro Studio enough over the last few days to test this. Will follow up early next week.

**Comment by hankinsoft** [comment]
No worries!

**Comment by jim256** [comment]
Still having issues with anything but the Snowflake browser query editor being slow. So in the interest of not keeping this open indefinitely, I'll close it for now. If it happens again, I'll be sure to let you know.

**Comment by jim256** [comment]
@hankinsoft I hate to say this, but I just saw it happen again:

<img width="59" alt="image" src="https://user-images.githubusercontent.com/15026225/68889034-54499600-06d9-11ea-9718-2a844dcc0838.png">

I'm not sure if it's been happening with other builds or not - this is just the first time that I noticed that it didn't correct itself after the query finished executing (it'll usually look funny while the query is executing, but then self correct when it finishes).

That being said, I think it's more rare now, and it always immediately corrects itself upon any horizontal scrolling. It doesn't really bother me and I hate to ask that it consume your time above other things, so I'm not personally going to re-open it for now.

macOS 10.15.1 (19B88)
SQLPro Studio 2019.54 (Build 110602.5)
Snowflake 3.51.1
